Please check whether you would consider to merge my changes

decarvalhoaa opened the pull request:
Hi Harasse,

I have been working on your fork and added a few things and corrected some bugs of the original 0.25, namely:

- Moved Email heanding and subject translation from __init__.php to Emalls.php, headings and subjects now use language files translation instead of using Polylang string translation, improved is_fully_refunded function and comment out filters in Endpoints.php that are creating issues with breadcrumb and menu links in My Account page
- Add translation of Shipping method and correct typo
- Document functions in Emails.php
- Correct \Taxonomies\ShippingClass typo
- Translate Payment gateway title & description at checkout, and bacs email instructions

let me know what you think.

Cheers
--decarvalhoaa

harasse commented:
Hi decarvalhoaa,

Your corrections are great and the coding is very good. I will take your 
fork for my own website (www.fanjamag.com). I started testing and I did 
not see a problem. I have not been able to test the translation of 
headings and susbjects emails because I have not yet understand how to 
set the "translation language files" (a add_filter gettext? By a .po / 
.mo file the plugin? help me please).

My conclusion is that it must be your fork which becomes the reference 
for all future developments. Because your developer expertise is clearly 
superior to mine. I suggest you also include the small correction 
Ebelrose 
(https://github.com/ebelrose/woo-poly-integration/commit/26c612c8446ee630b71aa6559c703366481a981c): 
I have not be able to test it, but it is probably relevant.

As we have no more news from Hyyan, my suggestion is that you should 
make a new wordpress plugin with a new name, to enable the whole 
community to use this code. I will help as I can to support your new 
plugin. Your plugin would of course recall in its heading that its 
origin is the Hyyan plugin. From my side, I'll try to join Hyyan to 
explain (but I have little chance of success because I see that for over 
9 months, he ceased all developer activities on Github).

What do you think of that?

Cheers
Harasse

Le 22/04/2016 22:14, decarvalhoaa a écrit :

> Hi Harasse,
> 
> I have been working on your fork and added a few things and corrected 
> some bugs of the original 0.25, namely:
> - Moved Email heanding and subject translation from _init_.php to
>   Emalls.php, headings and subjects now use language files
>   translation instead of using Polylang string translation, improved
>   is_fully_refunded function and comment out filters in
>   Endpoints.php that are creating issues with breadcrumb and menu
>   links in My Account page
> - Add translation of Shipping method and correct typo
> - Document functions in Emails.php
> - Correct \Taxonomies\ShippingClass typo
> - Translate Payment gateway title & description at checkout, and
>   bacs email instructions
> 
> let me know what you think.
> 
> Cheers
> --decarvalhoaa
> 
> ---
> 
> ```
>     You can view, comment on, or merge this pull request online at:
> ```
> 
> https://github.com/harasse/woo-poly-integration/pull/1
> 
> ```
>     Commit Summary
> ```
> - Moved Email heanding and subject translation from **init**.php to
>   Emails.php, headings and subjects now use language files
>   translation instead of using Polylang string translation, improved
>   is_fully_refunded function and comment out filters in
>   Endpoints.php that are creating issues with breadcrumb and menu
>   links in My Account page
> - Add translation of Shipping method and correct typo
> - Document functions in Emails.php
> - Correct \Taxonomies\ShippingClass typo
> - Translate Payment gateway title & description at checkout, and
>   bacs email instructions
>   
>   ```
>   File Changes
>   ```
> - _M_ **init**.php
>   https://github.com/harasse/woo-poly-integration/pull/1/files#diff-0
>   (215)
> - _M_ src/Hyyan/WPI/Emails.php
>   https://github.com/harasse/woo-poly-integration/pull/1/files#diff-1
>   (343)
> - _M_ src/Hyyan/WPI/Endpoints.php
>   https://github.com/harasse/woo-poly-integration/pull/1/files#diff-2
>   (6)
> - _M_ src/Hyyan/WPI/Gateways.php
>   https://github.com/harasse/woo-poly-integration/pull/1/files#diff-3
>   (56)
> - _A_ src/Hyyan/WPI/Gateways/GatewayBACS.php
>   https://github.com/harasse/woo-poly-integration/pull/1/files#diff-4
>   (108)
> - _D_ src/Hyyan/WPI/Taxonomies/ShippingCalss.php
>   https://github.com/harasse/woo-poly-integration/pull/1/files#diff-5
>   (29)
> - _A_ src/Hyyan/WPI/Taxonomies/ShippingClass.php
>   https://github.com/harasse/woo-poly-integration/pull/1/files#diff-6
>   (61)
> - _M_ src/Hyyan/WPI/Taxonomies/Taxonomies.php
>   https://github.com/harasse/woo-poly-integration/pull/1/files#diff-7
>   (2)
>   
>   ```
>   Patch Links:
>   ```
> - https://github.com/harasse/woo-poly-integration/pull/1.patch
> - https://github.com/harasse/woo-poly-integration/pull/1.diff
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub 
> https://github.com/harasse/woo-poly-integration/pull/1

decarvalhoaa commented:
Hi Harasse,

I'm happy you approve my suggestions.

Regarding the Email subjects and headings topic, in principle you shouldn't need to do anything. You just need to have the woocommerce .mo and .po files installed for the languages you want your site to support. I'm sure you have that already installed for EN and FR for your site. 
I thought this would be easier for the users, since it works out of the box, and they don't need to translate anything.
The only thing I'm doing, is to use filter hooks to wrap the subject and heading strings around the function __() like this 

`__( $subject, 'woocommerce' )`

These strings are static (defined in the Woocommerce Email admin settings) and Woocommerce does use __() in the Email class construct but this is too early and Polylang has not yet worked its magic. That is why the main body content of the email is in the order language but everything defined in the __construct is not.

I may publish my fixes in the wordpress.org site, but to be honest I'm a bit scare since I'm not a developer and like yourself don't have a lot of time. Actually this was the first time I used git. For sure I will continue to patch and enhance the plugin as I come across problems on my own website. But it may be that a big Wordpress, Woocommerce or Polylang upcoming release can break it completely and I don't have the skills to fix it and it may be that I need to find something else.

It would be great if you and the other folks that have been patching the plugin could still support and work on the plugin. 2 heads work better than 1 :-)

I'm in contact with some Plugin owners (free and premium) like Woocommerce PDF Invoices and Credit Notes to enable support for Polylang. I am getting a lot of help. I will try to do the same with Polylang and Woocommerce to understand upcoming changes and make the integration easier (like the email subject stuff).

I have lookup Ebelrose patch, and his/her changes are already applied in my local repo but not yet pushed upstream.

harasse commented:
Hi decarvalhoaa,

Great news: Hyyan is now in Germany and said he will continue to support 
and to update its plugin. Please, find below a copy of mails we 
exchanged. So it seems best that we store in your github fork the 
updates suggested by the community, waiting that Hyyan regain control of 
its plugin. Do you feel comfortable with that suggestion ?

Regarding the language translation files, I have problem to make it 
work. Mails from my French orders are OK, but the mail for English 
orders have neither the subject nor the title translated into English 
(the body is Ok). My setup has two languages: fr_FR (default) and en_US. 
The languages identified in my platform are:

wp-content/languages/fr_FR.po

wp-content/languages/fr_FR.mo

wp-content/languages/es_ES.po

wp-content/languages/es_ES.mo

wp-content/languages/de_DE.po

wp-content/languages/de_DE.mo

wp-content/languages/continents-cities-fr_FR.po

wp-content/languages/continents-cities-fr_FR.mo

wp-content/languages/continents-cities-es_ES.po

wp-content/languages/continents-cities-es_ES.mo

wp-content/languages/continents-cities-de_DE.po

wp-content/languages/continents-cities-de_DE.mo

wp-content/languages/admin-network-fr_FR.po

wp-content/languages/admin-network-fr_FR.mo

wp-content/languages/admin-network-es_ES.po

wp-content/languages/admin-network-es_ES.mo

wp-content/languages/admin-network-de_DE.po

wp-content/languages/admin-network-de_DE.mo

wp-content/languages/admin-fr_FR.po

wp-content/languages/admin-fr_FR.mo

wp-content/languages/admin-es_ES.po

wp-content/languages/admin-es_ES.mo

wp-content/languages/admin-de_DE.po

wp-content/languages/admin-de_DE.mo

wp-content/themes/hustle/lang/en_GB.po

wp-content/languages/woocommerce/woocommerce-de_DE.po

wp-content/languages/woocommerce/woocommerce-de_DE.mo

wp-content/languages/woocommerce/woocommerce-admin-de_DE.po

wp-content/languages/woocommerce/woocommerce-admin-de_DE.mo

wp-content/languages/plugins/wp-optimize-fr_FR.po

wp-content/languages/plugins/wp-optimize-fr_FR.mo

wp-content/languages/plugins/wp-optimize-de_DE.po

wp-content/languages/plugins/wp-optimize-de_DE.mo

wp-content/languages/plugins/woocommerce-fr_FR.po

wp-content/languages/plugins/woocommerce-fr_FR.mo

wp-content/languages/plugins/woocommerce-es_ES.po

wp-content/languages/plugins/woocommerce-es_ES.mo

wp-content/languages/plugins/woocommerce-de_DE.po

wp-content/languages/plugins/woocommerce-de_DE.mo

wp-content/languages/plugins/woocommerce-admin-fr_FR.po

wp-content/languages/plugins/woocommerce-admin-fr_FR.mo

wp-content/languages/plugins/polylang-fr_FR.po

wp-content/languages/plugins/polylang-fr_FR.mo

wp-content/languages/plugins/polylang-de_DE.po

wp-content/languages/plugins/polylang-de_DE.mo

wp-content/languages/plugins/enhanced-media-library-fr_FR.po

wp-content/languages/plugins/enhanced-media-library-fr_FR.mo

wp-content/languages/plugins/backwpup-fr_FR.po

wp-content/languages/plugins/backwpup-fr_FR.mo

wp-content/languages/plugins/backwpup-de_DE.po

wp-content/languages/plugins/backwpup-de_DE.mo

wp-content/languages/plugins/akismet-fr_FR.po

wp-content/languages/plugins/akismet-fr_FR.mo

Is the issue coming from that I choose en_US, which is the standard 
language of wordpress, as the recipient of language translation?

Moreover I do not understand how a title or a subject changed in the 
WooCommerce settings could be translated automatically. Your advice 
would be helpful.

Regards

Jacques Harasse (jharasse@gmail.com mailto:jharasse@gmail.com)

---

Le 24/04/2016 14:49, Jacques Harasse a écrit :

```
Hi Hyyan,

It is great news that you said me there. To be frank, I suspected
that your situation in Syria was very difficult and I even imagined
the worst. I'm glad you finally are in Germany which do a lot for
the reception of refugees.
Finally it is therefore better not to launch our new plugin, and to
wait for your situation be stabilized in Germany. Thus, to allow you
to resume the support of your plugin when you will be ok, we can
build up in the Github fork @decarvalhoaa the updates suggested by
the community.

Personally and friendly, if I can be of any help from France, do not
hesitate to tell me.

It is really good news.
Regards
Jacques Harasse



Le 24/04/2016 13:34, Hyyan AF a écrit :

Hello Harassse

I would like to thank you for telling me,  It would be great idea at
the moment to Support the community.

I would like too to tell you that I have not stopped the support for
my plugin, I have already worked on some updates and fixes. To
improve the plugin. 

I was not able to respond in the last few months or offer Support
because I was forced to leave my Home Syria aś many,  Now I am
living in Germany "Refugge" but my situation is much better now,
probably I will be able to offer support and push fixes and Updates
to my plugin soon.

Best Regrads

Von meinem Sony Xperia™-Smartphone gesendet



---- Jacques Harasse schrieb ----

Hi Hyyan,

woo-poly-integration is a great plugin that many consider the
richest and most successful to integrate WooCommerce with Polylang.
Indeed many people use the plugin on their production site. You
stopped the support and the development of the plugin and the
community is very sad. It is your choice and we respect it.

Updates in Polylang, Wordpress and WooCommerce led some to forks in
Github to suggest changes. A handful of us agreed to gather the
changes proposed in a new wordpress plugin to be more easily used by
the community. @decarvalhoaa will manage this new plugin for the
community.

This new plugin will appear as an extension of woo-poly-integration
and of your work. Hyyan, be assured that you will always be welcome
if you choose to bring your expertise and advice in this project.

Regards

Harasse
```

Le 23/04/2016 13:23, decarvalhoaa a écrit :

> Hi Harasse,
> 
> I'm happy you approve my suggestions.
> 
> Regarding the Email subjects and headings topic, in principle you 
> shouldn't need to do anything. You just need to have the woocommerce 
> .mo and .po files installed for the languages you want your site to 
> support. I'm sure you have that already installed then in EN and FR 
> for your site.
> I thought this would be easier for the users, since it works out of 
> the box, and they don't need to translate anything.
> The only thing I'm doing, is to use filter hooks to wrap the subject 
> and heading strings around __() like this
> 
> |__( $subject, 'woocommerce' )|
> 
> This strings are static (defined in the Woocommerce Email admin 
> settings) and Woocommerce does use __() in the Email class construct 
> but this is too early and Polylang was not yet worked it's magic. That 
> is why the main body content of the email is in the order language but 
> everything defined in the __construct is not.
> 
> I may publish my fixes in the wordpress.org site, but to be honest I'm 
> a bit scare since I'm not a developer and like yourself don't have a 
> lot of time. Actually this was the even the first time I used git. For 
> sure I will continue to patch and enhance the plugin as I come across 
> problems on my own website. But it may be that a big Wordpress, 
> Woocommerce or Polylang upcoming release can break it completely and I 
> don't have the skills to fix it and it may be that I need to find 
> something else.
> 
> It would be great if you and the other folks that have been patching 
> the plugin could still support and work on the plugin. 2 heads work 
> better than 1 :-)
> 
> I'm in contact with some Plugin owners (free and premium) like 
> Woocommerce PDF Invoices and Credit Notes to enable support for 
> Polylang. I am getting a lot of help. I will try to do the same with 
> Polylang and Woocommerce to understand upcoming changes and make the 
> integration easier (like the email subject stuff).
> 
> I have lookup Ebelrose patch, and he/she changes are already applied 
> in my local repo but not yet pushed upstream.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> https://github.com/harasse/woo-poly-integration/pull/1#issuecomment-213722798

decarvalhoaa commented:
Hi Harasse,

Your mail is full of good news. Its fantastic that Hyyan made it to 
Germany. I'm based in Germany, near Düsseldorf, and would love to help. 
Feel free to pass my contact.

Second good news is that Hyyan will continue to support its plugin. This 
is great and definitely we should park the idea to publish my fork. 
Instead I will contribute to Hyyn work with testing, patches and perhaps 
some feature requests.

Regarding the Email Subjects and Headings, it seems that my approach is 
not very smart after all. I suspect that my solution doesn't work if the 
Default Language is not EN. Sorry for the inconvenient. I need to look 
into this over the week.
I based my solution on my use case, i.e. my default language is EN and 
my second language DE. When I want to have a custom translation I edit 
the op files using the Loco plugin.

Cheers
A

On 04/24/2016 03:56 PM, harasse wrote:

> Hi decarvalhoaa,
> 
> Great news: Hyyan is now in Germany and said he will continue to support
> and to update its plugin. Please, find below a copy of mails we
> exchanged. So it seems best that we store in your github fork the
> updates suggested by the community, waiting that Hyyan regain control of
> its plugin. Do you feel comfortable with that suggestion ?
> 
> Regarding the language translation files, I have problem to make it
> work. Mails from my French orders are OK, but the mail for English
> orders have neither the subject nor the title translated into English
> (the body is Ok). My setup has two languages: fr_FR (default) and en_US.
> The languages identified in my platform are:
> 
> wp-content/languages/fr_FR.po
> 
> wp-content/languages/fr_FR.mo
> 
> wp-content/languages/es_ES.po
> 
> wp-content/languages/es_ES.mo
> 
> wp-content/languages/de_DE.po
> 
> wp-content/languages/de_DE.mo
> 
> wp-content/languages/continents-cities-fr_FR.po
> 
> wp-content/languages/continents-cities-fr_FR.mo
> 
> wp-content/languages/continents-cities-es_ES.po
> 
> wp-content/languages/continents-cities-es_ES.mo
> 
> wp-content/languages/continents-cities-de_DE.po
> 
> wp-content/languages/continents-cities-de_DE.mo
> 
> wp-content/languages/admin-network-fr_FR.po
> 
> wp-content/languages/admin-network-fr_FR.mo
> 
> wp-content/languages/admin-network-es_ES.po
> 
> wp-content/languages/admin-network-es_ES.mo
> 
> wp-content/languages/admin-network-de_DE.po
> 
> wp-content/languages/admin-network-de_DE.mo
> 
> wp-content/languages/admin-fr_FR.po
> 
> wp-content/languages/admin-fr_FR.mo
> 
> wp-content/languages/admin-es_ES.po
> 
> wp-content/languages/admin-es_ES.mo
> 
> wp-content/languages/admin-de_DE.po
> 
> wp-content/languages/admin-de_DE.mo
> 
> wp-content/themes/hustle/lang/en_GB.po
> 
> wp-content/languages/woocommerce/woocommerce-de_DE.po
> 
> wp-content/languages/woocommerce/woocommerce-de_DE.mo
> 
> wp-content/languages/woocommerce/woocommerce-admin-de_DE.po
> 
> wp-content/languages/woocommerce/woocommerce-admin-de_DE.mo
> 
> wp-content/languages/plugins/wp-optimize-fr_FR.po
> 
> wp-content/languages/plugins/wp-optimize-fr_FR.mo
> 
> wp-content/languages/plugins/wp-optimize-de_DE.po
> 
> wp-content/languages/plugins/wp-optimize-de_DE.mo
> 
> wp-content/languages/plugins/woocommerce-fr_FR.po
> 
> wp-content/languages/plugins/woocommerce-fr_FR.mo
> 
> wp-content/languages/plugins/woocommerce-es_ES.po
> 
> wp-content/languages/plugins/woocommerce-es_ES.mo
> 
> wp-content/languages/plugins/woocommerce-de_DE.po
> 
> wp-content/languages/plugins/woocommerce-de_DE.mo
> 
> wp-content/languages/plugins/woocommerce-admin-fr_FR.po
> 
> wp-content/languages/plugins/woocommerce-admin-fr_FR.mo
> 
> wp-content/languages/plugins/polylang-fr_FR.po
> 
> wp-content/languages/plugins/polylang-fr_FR.mo
> 
> wp-content/languages/plugins/polylang-de_DE.po
> 
> wp-content/languages/plugins/polylang-de_DE.mo
> 
> wp-content/languages/plugins/enhanced-media-library-fr_FR.po
> 
> wp-content/languages/plugins/enhanced-media-library-fr_FR.mo
> 
> wp-content/languages/plugins/backwpup-fr_FR.po
> 
> wp-content/languages/plugins/backwpup-fr_FR.mo
> 
> wp-content/languages/plugins/backwpup-de_DE.po
> 
> wp-content/languages/plugins/backwpup-de_DE.mo
> 
> wp-content/languages/plugins/akismet-fr_FR.po
> 
> wp-content/languages/plugins/akismet-fr_FR.mo
> 
> Is the issue coming from that I choose en_US, which is the standard
> language of wordpress, as the recipient of language translation?
> 
> Moreover I do not understand how a title or a subject changed in the
> WooCommerce settings could be translated automatically. Your advice
> would be helpful.
> 
> Regards
> 
> Jacques Harasse (jharasse@gmail.com mailto:jharasse@gmail.com)
> 
> ---
> 
> Le 24/04/2016 14:49, Jacques Harasse a écrit :
> 
> Hi Hyyan,
> 
> It is great news that you said me there. To be frank, I suspected
> that your situation in Syria was very difficult and I even imagined
> the worst. I'm glad you finally are in Germany which do a lot for
> the reception of refugees.
> Finally it is therefore better not to launch our new plugin, and to
> wait for your situation be stabilized in Germany. Thus, to allow you
> to resume the support of your plugin when you will be ok, we can
> build up in the Github fork @decarvalhoaa the updates suggested by
> the community.
> 
> Personally and friendly, if I can be of any help from France, do not
> hesitate to tell me.
> 
> It is really good news.
> Regards
> Jacques Harasse
> 
> Le 24/04/2016 13:34, Hyyan AF a écrit :
> 
> Hello Harassse
> 
> I would like to thank you for telling me, It would be great idea at
> the moment to Support the community.
> 
> I would like too to tell you that I have not stopped the support for
> my plugin, I have already worked on some updates and fixes. To
> improve the plugin.
> 
> I was not able to respond in the last few months or offer Support
> because I was forced to leave my Home Syria aś many, Now I am
> living in Germany "Refugge" but my situation is much better now,
> probably I will be able to offer support and push fixes and Updates
> to my plugin soon.
> 
> Best Regrads
> 
> Von meinem Sony Xperia™-Smartphone gesendet
> 
> ---- Jacques Harasse schrieb ----
> 
> Hi Hyyan,
> 
> woo-poly-integration is a great plugin that many consider the
> richest and most successful to integrate WooCommerce with Polylang.
> Indeed many people use the plugin on their production site. You
> stopped the support and the development of the plugin and the
> community is very sad. It is your choice and we respect it.
> 
> Updates in Polylang, Wordpress and WooCommerce led some to forks in
> Github to suggest changes. A handful of us agreed to gather the
> changes proposed in a new wordpress plugin to be more easily used by
> the community. @decarvalhoaa will manage this new plugin for the
> community.
> 
> This new plugin will appear as an extension of woo-poly-integration
> and of your work. Hyyan, be assured that you will always be welcome
> if you choose to bring your expertise and advice in this project.
> 
> Regards
> 
> Harasse
> 
> Le 23/04/2016 13:23, decarvalhoaa a écrit :
> 
> > Hi Harasse,
> > 
> > I'm happy you approve my suggestions.
> > 
> > Regarding the Email subjects and headings topic, in principle you
> > shouldn't need to do anything. You just need to have the woocommerce
> > .mo and .po files installed for the languages you want your site to
> > support. I'm sure you have that already installed then in EN and FR
> > for your site.
> > I thought this would be easier for the users, since it works out of
> > the box, and they don't need to translate anything.
> > The only thing I'm doing, is to use filter hooks to wrap the subject
> > and heading strings around __() like this
> > 
> > |__( $subject, 'woocommerce' )|
> > 
> > This strings are static (defined in the Woocommerce Email admin
> > settings) and Woocommerce does use __() in the Email class construct
> > but this is too early and Polylang was not yet worked it's magic. That
> > is why the main body content of the email is in the order language but
> > everything defined in the __construct is not.
> > 
> > I may publish my fixes in the wordpress.org site, but to be honest I'm
> > a bit scare since I'm not a developer and like yourself don't have a
> > lot of time. Actually this was the even the first time I used git. For
> > sure I will continue to patch and enhance the plugin as I come across
> > problems on my own website. But it may be that a big Wordpress,
> > Woocommerce or Polylang upcoming release can break it completely and I
> > don't have the skills to fix it and it may be that I need to find
> > something else.
> > 
> > It would be great if you and the other folks that have been patching
> > the plugin could still support and work on the plugin. 2 heads work
> > better than 1 :-)
> > 
> > I'm in contact with some Plugin owners (free and premium) like
> > Woocommerce PDF Invoices and Credit Notes to enable support for
> > Polylang. I am getting a lot of help. I will try to do the same with
> > Polylang and Woocommerce to understand upcoming changes and make the
> > integration easier (like the email subject stuff).
> > 
> > I have lookup Ebelrose patch, and he/she changes are already applied
> > in my local repo but not yet pushed upstream.
> > 
> > —
> > You are receiving this because you commented.
> > Reply to this email directly or view it on GitHub
> > 
> > https://github.com/harasse/woo-poly-integration/pull/1#issuecomment-213722798
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> https://github.com/harasse/woo-poly-integration/pull/1#issuecomment-213961913

decarvalhoaa commented:
Hi Harasse,

I reworked the Email translation. Now all settings are registered in the Polylang Strings Translation table.
In case the Admin User has changed the default subjects/headings in WooCommerce -> Settings -> Emails (not pressed Save) the settings can't be retreived from the database with get_option. Therefore, I added a default, same was used by WooCommerce.

There were also some settings missing like subject_ / heading_downloadable etc that I added.

Do you mind testing this?

Cheers,
decarvalhoaa
